evm_loader/program/src/precompile_contracts.rs

```diff
@@ -280,6 +286,70 @@ pub fn erc20_wrapper<'a, B: AccountStorage>(
     }
 }
 
+// Neon token method ids:
+//--------------------------------------------------
+// withdraw(bytes32)           => 8e19899e
+//--------------------------------------------------
+const NEON_TOKEN_METHOD_WITHDRAW_ID: &[u8; 4]       = &[0x8e, 0x19, 0x89, 0x9e];
+
+/// Call inner `neon_token`
+#[must_use]
+#[allow(clippy::too_many_lines)]
+pub fn neon_token<'a, B: AccountStorage>(
+    input: &[u8],
+    context: &evm::Context,
+    state: &mut ExecutorState<'a, B>
+)
+    -> Capture<(ExitReason, Vec<u8>), Infallible>
+{
+    debug_print!("neon_token({})", hex::encode(&input));
+
+    let (method_id, rest) = input.split_at(4);
+    let method_id: &[u8; 4] = method_id.try_into().unwrap_or(&[0_u8; 4]);
+    let min_amount: u64 = u64::pow(10, u32::from(token_mint::decimals()));
+
+    if method_id == NEON_TOKEN_METHOD_WITHDRAW_ID  {
+        if state.metadata().is_static() {
+            let revert_message = b"neon_token: withdraw is not allowed in static context".to_vec();
+            return Capture::Exit((ExitReason::Revert(evm::ExitRevert::Reverted), revert_message))
+        }
+
+        let source = context.address; // caller contract
+
+        // owner of the associated token account
+        let destination = array_ref![rest, 0, 32];
+        let destination = Pubkey::new_from_array(*destination);
+
+        let (spl_amount, remainder) =
+            context
+            .apparent_value
+            .div_mod(U256::from(min_amount));
+
+        if spl_amount > U256::from(u64::MAX) {
+            let revert_message = b"neon_token: transfer amount exceeds maximum".to_vec();
+            return Capture::Exit((ExitReason::Revert(evm::ExitRevert::Reverted), revert_message))
+        }
+
+        if remainder.as_u64() != 0 {
```

Comment:
Do you really need to convert it to u64 to compare with zero?
`!remainder.is_zero()`